### DISCUSSION TITLE
Simulation Mode

### DISCUSSION THREAD

**Discussion by alu0100770032** [asreview/asreview#986] [Q&A] [answered]
Hello ASreview community, I am a Computer Science student working on my final graduate work. This work is about tools for systematic reviews.
I intend to build a small machine learning app that categorize as simple as possible and then compare to the super good ASreview.
But the documentation is a bit obscure around the simulation mode.
Considering one of your own already reviewed sample filtes Appenzeller-Herzog_2020 this one for example. (3453 studies, 29 relevant, the rest irrelenvant)
I want to compare how good and close to the proper papers it comes if I feed the algorythm  a % or some of the relevant and see how many come as relevant , how close to the other relevant it comes and so on, and then compare it to mine.
Considering that backgroud, I dont know to simulate that with ASreview, I thought it was throught the simulation mode using the config file, but there is no documentation or examples on how to do that.
How do I load a whole file then feed it with a given relevant studies out of the file , and obtain back the list of the relevant ones by the algorythm?
Btw excuse my english its not my mother lang, thank you in advance.

**Comment by Rensvandeschoot** [comment, marked as answer]
Great you are interested to use ASReview for your work! 

Concerning the simulation mode, you can find a tutorial [on the blog](https://asreview.nl/blog/simulation-mode-class-101/). 
All options for the simulation mode are described on [ReadTheDocs](https://asreview.readthedocs.io/en/latest/API/cli.html#simulate). You are probably interested is selecting specific prior records, see for an example [this `.sh` file](https://github.com/asreview/paper-guidelines-KIFMS/blob/main/jobs_run_simulation.sh). One line used in this study shows how to specify specific records: 

```
asreview simulate data[...].csv -s output[...].h5 --prior_record_id 0 115 176 151 37 13 164 162 114 32 130 --seed 165
```

There is also [a plugin](https://github.com/asreview/asreview-statistics) to obtain all kinds of statistics, including the average time to discovery (ATD), see also [this paper](https://osf.io/w6qbg/). This metric can be used to find out which paper is most difficult to find, see for an application[ this repository](https://github.com/asreview/paper-depression-brouwer-simulation-ATD) .

Recall plots can be obtained via [this plug-in](https://github.com/asreview/asreview-visualization). 

Hope this helped!

**Comment by alu0100770032** [comment]
Good morning and thank you for the response! I am sorry I took so long to answer back again.
As I am slowly learning how all this works, as I stated my background, and my objective with  using and testing ASReview, it seems maybe the API suits me better. But I am having trouble as well making it work.
The simulation mode it gives stats based on a review already completed and gives me the advantages and time saved if fully manual system would have been used( if I understand this correctly) but I wanted to know deeper, like to know the ranking system given out of a number of recommendations on a nonfully reviewed data. Like learning more on the machine learning process and test the machine learning process coming from a CS point.

Lets say I want to do this:
I have 100 studies. And I know the 1-10 are the relevant ones. ( but are not clasified yet as such in the data)  
I feed the ASReview study N 2 out of the 100 as relevant,  or 2 and 3, or 3 and 5 . 
Then I want to know how accurate and how good is showing me 1, 4, 6,7,8,9 and 10 how do ASReview rank them and so on
Sorry if I look like ranting I am getting to know about reviews as my work moves on.
Really thank you for the previous response, and it took me a time to know that it was close to what I needed but not exactly. (Unless I probably got something completely wrong, again which is likely)

**Reply by Rensvandeschoot** [reply]
no problem! happy to help you!

ASReview exports the rank order of the unlabelled records, [see RTDs](https://asreview.readthedocs.io/en/latest/features/post_screening.html?highlight=export#download-results). Is this what you need?
